WordPress/Classes/ViewRelated/Site Creation/Web Address/DesignSystem/Components/Modifiers/TextStyle.swift

```diff
@@ -0,0 +1,16 @@
+enum TextStyle {
+    case heading1
+    case heading2
+    case heading3
+    case heading4
+    case bodySmall(Weight)
+    case bodyMedium(Weight)
+    case bodyLarge(Weight)
+    case footnote
+    case caption
+
+    enum Weight {
+        case regular
+        case emphasized
+    }
+}
```

Comment:
Have you considered making this a `struct` rather than an `enum` (like `DSButtonStyle` does)? 

This way, consumers would not have to explicitly or implicitly know about all the possible cases, but would "simply" read the configuration that each style has.

Rough example based on the usages I could see in this diff in `UILabel` and `Text`

```swift
struct TextStyle {
  let case: Text.Case?
  let font: Font
  let uiFont: UIFont
}

extension TextStyle {

  static let heading1 = TextStyle(case: nil, font:  Font.DS.heading1, uiFont: UIFont.DS.heading1)
  static let heading2 = ...
}
```

Unless I made a mistake, both `ViewBuilders` on `Text` and `UILabel` would work in the same way.

Another advantage of using `structs` instead of `enums` for configurations is that all the information for what the configuration is lives in the instance, as opposed to being spread across multiple switch statements. 

Additionally, adding and removing configurations "cases" consists of adding or removing a `static let` property, as opposed to modify a number of switch statements.

For more on the benefits of `structs` over `enums` for this kind of configurations, see https://www.jessesquires.com/blog/2016/07/31/enums-as-configs/

Reply:
Maybe I am missing your point but inflexibility is precisely what I was aiming for here. I wish to limit the usage and not allow combinations of different use-cases like so:
`TextStyle(case: .uppercase, font: Font.DS.heading1)`

The reason is that, if the API allows customisation, devs can introduce new styles without being aware that, that is indeed a new style that hadn't exist in the Design System before.

Scenario:
1. Designer adds a new font & case pair my mistake where they actually intended to use an existing, pre-defined pair.
2. Developer uses the struct with the specified font and case.
3. The API is not limiting so allows such a combination to be created.

However when we use an enum, it will require a deliberate addition. If in fact the designer wanted us to add such a new pair, then we can go ahead and do that.

Does that make sense? Or perhaps I missed your point.

Reply:
The difference with `DSButtonStyle` is that, every possible combination is supported by the Design System (that is on Figma). So mix and match of those enums are not a problem as we want all the combinations.

Reply:
@alpavanoglu 

> Maybe I am missing your point but inflexibility is precisely what I was aiming for here. I wish to limit the usage and not allow combinations of different use-cases like so: `TextStyle(case: .uppercase, font: Font.DS.heading1)`

Would that be possible by making closing `TextStyle`? That is, by making its `init` `private`?

> However when we use an enum, it will require a deliberate addition. If in fact the designer wanted us to add such a new pair, then we can go ahead and do that.
>
> Does that make sense? Or perhaps I missed your point.

Yes it makes sense. And I think that an approach where `TextStyle` has `private` initializer would achieve the same result. That is, adding a new style would require a deliberate PR at the DesignSystem module level.

_Anyways, this is not a blocker for me. I consider having to navigate multiple `switch` to add/remove/update entries clunky, but I am also aware that there shouldn't be many such changes in the future. I'd prefer not to use `enum`s for configuration from an architectural point of view, but I won't push against this anymore if you feel the current  setup serves the code better._

Reply:
I do understand and share your concerns regarding enums. When one adds a new case, Xcode goes all hysterical, raising 17 errors :-] 

> Would that be possible by making closing TextStyle? That is, by making its init private?
How would the caller then use the `func style(_ style: TextStyle) -> some View {` function if they cannot instantiate a `style`?